agentlightning/llm_proxy.py

```diff
@@ -519,93 +523,110 @@ async def dispatch(self, request: Request, call_next: Callable[[Request], Awaita
         return response
 
 
-class MessageInspectionMiddleware:
+class MessageInspectionMiddleware(BaseHTTPMiddleware):
+    """Middleware to inspect the request and response bodies.
 
-    def __init__(self, app):
-        self.app = app
-
-    async def __call__(self, scope, receive, send):
-        if scope["type"] != "http":
-            await self.app(scope, receive, send)
-            return
+    It's for debugging purposes. Add it via "message_inspection" middleware alias.
+    """
 
-        print(f">>> DEBUG: Received request with scope\n{scope}\n")
+    async def dispatch(self, request: Request, call_next: Callable[[Request], Awaitable[Response]]) -> Response:
+        ti = time.time()
+        logger.info(f"Received request with scope: {request.scope}")
+        logger.info(f"Received request with body: {await request.body()}")
+        response = await call_next(request)
+        elapsed = time.time() - ti
+        logger.info(f"Response to request took {elapsed} seconds")
+        logger.info(f"Received response with status code: {response.status_code}")
+        logger.info(f"Received response with body: {response.body}")
+        return response
 
-        async def log_request():
```

Comment:
This does not print request body. The content have different types of messages, including `message_start`, `message_stop`, `content_block_start`, `content_block_delta`, `content_block_stop`. They are used to indicate the stream chunks and some metadata. `anthropic_stream_generator` shows the format for each type of message, and I suppose this is different from request body and response.